docs/developer-experience/readme.md

```diff
@@ -0,0 +1,231 @@
+- it returns a value or a promise that would eventuall return a value.
+
+#### Why it is so important?
+
+Because not only the application code uses this as a type. Bauta.js itself uses this signature for its internal execution. Even bauta.js more advanced abstractions are in turn or return a StepFunction. And finally, because almost all the logic defined in terms of endpoints are a combination of one or more StepFunctions (check pipeline for more details).
+
+One could Think about the code of a bauta.js application as a bunch of StepFunctions that are run one after another or in parallel and whose results are passed from one to the next defined in your code. 
+
+#### What are the main parts of an StepFunction?
+
+These are three input parameters of an StepFunction:
+
+- prev: is the previous value. this always take the value (or values) returned by the previous StepFunction. This value can be undefined, usually because is the the first StepFunction and there is none previously.
+
+- context: this is the context of the request being processed. You can think of this parameter as the session of the request [^2].
+
+- bautaJSInstance: this object represents the bauta.js server instance itself. The most used field of this object is called staticConfig and has all the variables loaded by bauta.js at startup time.
+
+### Operation 
+
+An operation is a type that abstracts a route definition. Initially, you may think of an Operation as the concept that glues together:
+- a route or endpoint defined in your swagger
+- if applicable, an schema that the operation has to conform to
+- a business logic that handles the input from the request and generates a response to the caller together with any possible side effect
+
+
+### Resolver 
+
+Operations are usually defined together in a single file which is called resolver in bauta.js. Thus each entity can have its own resolver that logically aggregates related operations.
+
+
+
+## An example 
+
+Let's try to make an example to help understand these concepts. We want to implement an endpoint that calculates how much money driving between two points would cost. The business logic of our endpoint should do the following:
+
+- from the incoming request call get the starting and destination points, and generate a request input to a service similar to Google's Distance Matrix API
+- send that request to the service
+- obtain the distance from the response from the previous request
+- using that distance, call another service that returns the price in fuel that you would use
+- return that value to the caller as the response of the incoming request
+
+### bauta.js code structure
+
+In bauta.js what we like to do is this:
+
+- define a resolver that has the operation defined and that setups as its handler a pipeline.
+- define a pipeline with a list of the steps required.
+
+If you are reading this for the first time, it may seem hard to grasp what it is, but at code level is just he following:
```

Comment:
```suggestion
In bauta.js, what we like to do is this:

- define a resolver that has the operation defined and that setups as its handler a pipeline.
- define a pipeline with a list of the steps required.

If you are reading this for the first time, it may seem complicated to grasp what it is, but on the code is just the following:
```